SingularityUI/app/views/taskHealthcheckNotificationSubview.coffee

```diff
@@ -45,11 +57,19 @@ class taskHealthcheckNotificationSubview extends View
         taskHealthyMessage: taskHealthyMessage
         hasSuccessfulHealthcheck: @model.get('healthcheckResults')?.length > 0 and _.find(@model.get('healthcheckResults'), (item) -> item.statusCode is 200)
         lastHealthcheckFailed: @model.get('healthcheckResults')?.length > 0 and @model.get('healthcheckResults')[0].statusCode isnt 200
+        healthcheckFailureReasonMessage: @healthcheckFailureReasonMessage()
         synced:           @model.synced
         config:           config
         tooManyRetries: @model.get('healthcheckResults').length > maxRetries and maxRetries != 0
         numberFailed: @model.get('healthcheckResults').length
         secondsElapsed: healthTimeoutSeconds
+        doNotDisplayHealthcheckNotification: deployFailureKilledTask
+
+    healthcheckFailureReasonMessage: () ->
```

Comment:
I named that healthcheckFailureReasonMessage because I figured if we wanted to add something else in there later we could do so in that function rather than have the logic spread around. For now, yes, it only looks for connection refused.

I will modify the message to make it more clear.